### PR TITLE
isync: add post-1.5.0 commits

### PR DESCRIPTION
**Pull request by Jdogzz**
Adds post-1.5.0 commits to fix breaking changes. See for example: https://bugzilla.redhat.com/show_bug.cgi?id=2314719

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by sagehane** [comment]
I'm also suffering from https://sourceforge.net/p/isync/bugs/77/ so I would like to see this merged.

File: pkgs/tools/networking/isync/default.nix [hunk 1]
```diff
@@ -1,32 +1,69 @@
...
+  nativeBuildInputs = [
+    autoreconfHook
+    pkg-config
+    perl
+  ] ++ lib.optionals withCyrusSaslXoauth2 [ makeWrapper ];
+  buildInputs =
+    with perl538Packages;
+    [
+      TimeDate
+    ]
```
Comment (felipeqq2) [inline pkgs/tools/networking/isync/default.nix:50]: I'd just use `buildInputs = [ perl538Packages.TimeDate other-packages ];`.

File: pkgs/tools/networking/isync/default.nix [hunk 2]
```diff
@@ -1,32 +1,69 @@
...
+  # Disabled by default as XOAUTH2 is an "OBSOLETE" SASL mechanism and this relies
+  # on a package that isn't really maintained anymore:
+  withCyrusSaslXoauth2 ? false,
+  cyrus-sasl-xoauth2,
+  makeWrapper,
 }:
 
 stdenv.mkDerivation (finalAttrs: {
   pname = "isync";
   version = "1.5.0";
```
Comment (felipeqq2) [inline pkgs/tools/networking/isync/default.nix:23]: This should be `1.5.0-unstable-2024-09-29`  (see https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#versioning).

**Review by felipeqq2 — APPROVED** [review]
Works fine, thanks for the PR! The commit history is a bit messed up though, could you squash all commits into one or two descriptive commits?

**Comment by Jdogzz** [comment]
> Works fine, thanks for the PR! The commit history is a bit messed up though, could you squash all commits into one or two descriptive commits?

Thanks for the feedback, all changes are now made and the commits have been squashed.

**Review by felipeqq2 — APPROVED** [review]

**Review by Moredread — APPROVED** [review]

**Comment by Moredread** [comment]
It'd be nice to have those fixes included as there are a few breaking changes without them.

**Pull request merged**